src/manifest.c

```diff
@@ -327,6 +326,20 @@ static struct manifest *manifest_from_file(int version, char *component, bool he
 		c = c2;
 
 		file->filename = strdup_or_die(c);
+		file->bundlename = strdup_or_die(c);
+		/* If the file is an iterative manifest, copy the from_version and rename the
+		 * manifest so it matches the bundle name */
+		if (file->is_iterative == 1) {
+			char *ptr;
+			ptr = strrchr(file->filename, '.');
+			if (ptr != NULL) {
+				file->from_version = atoi(ptr + 1);
+			}
+			ptr = strrchr(file->bundlename, 'I');
+			if (ptr != NULL) {
+				*(ptr - 1) = '\0';
```

Comment:
Leave a comment here that you need to set `ptr - 1` to `\0` because of the  `.` before the `I`